### PR TITLE
feat(clerk-js,shared): getMemberships support for paginated responses + useOrganization adds `memberships`

### PR DESCRIPTION
**Pull request by panteliselef**
## Type of change

- [ ] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

## Packages affected

- [x] `@clerk/clerk-js`
- [ ] `@clerk/clerk-react`
- [ ] `@clerk/nextjs`
- [ ] `@clerk/remix`
- [ ] `@clerk/types`
- [ ] `@clerk/themes`
- [ ] `@clerk/localizations`
- [ ] `@clerk/clerk-expo`
- [ ] `@clerk/backend`
- [ ] `@clerk/clerk-sdk-node`
- [x] `@clerk/shared`
- [ ] `@clerk/fastify`
- [ ] `@clerk/chrome-extension`
- [ ] `gatsby-plugin-clerk`
- [ ] `build/tooling/chore`

## Description

- [x] `npm test` runs as expected.
- [x] `npm run build` runs as expected.
- [x] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports

### CODE REVIEW

**Review by dimkl — APPROVED** [review]

File: .changeset/hot-chairs-refuse.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@clerk/shared': patch
+---
+
+Deprecate membershipList in favor of `memberships` that supports paginated responses
```
Comment (dimkl) [inline .changeset/hot-chairs-refuse.md:5]: 🙃 could you decorate `membershipList` with backticks like you did with `membershipts` word?

File: packages/types/src/organization.ts [hunk 2]
```diff
@@ -60,10 +60,26 @@ export interface OrganizationResource extends ClerkResource {
...
+/**
+ * @deprecated use GetMembersParams
+ */
 export type GetMembershipsParams = {
   role?: MembershipRole[];
 } & ClerkPaginationParams;
 
+export type GetMembersParams = {
+  /**
+   * This the starting point for your fetched results. The initial value persists between re-renders
```
Comment (desiprisg) [inline packages/types/src/organization.ts:72]: Did you mean `This is the starting point` ?

**Comment by panteliselef** [comment]
> Does it make sense to add any tests for this change?

@desiprisg there are no test for useOrganization at this moment. I will be adding them in a separate PR as it needs to cover all functionalities of the hook

**Comment by clerk-cookie** [comment]
This PR has been automatically locked since there has not been any recent activity after it was closed. Please open a new issue for related bugs.

**Pull request merged**